src/Type/Php/StrlenFunctionReturnTypeExtension.php

```diff
@@ -30,8 +33,16 @@ public function getTypeFromFunctionCall(
 		}
 
 		$argType = $scope->getType($args[0]->value);
+
+		if ($argType instanceof ConstantStringType) {
+			return new ConstantIntegerType(strlen($argType->getValue()));
+		}
+		if ($argType instanceof BooleanType) {
+			return IntegerRangeType::fromInterval(0, 1);
+		}
+
 		$isNonEmpty = $argType->isNonEmptyString();
-		if ($isNonEmpty->yes()) {
+		if ($isNonEmpty->yes() || $argType instanceof IntegerType) {
```

Comment:
Please avoid using `instanceof *Type`. This article talks about why: https://phpstan.org/developing-extensions/type-system